### PR TITLE
Common files feature

### PR DESCRIPTION
**Pull request by briangsweeney**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesMetaDataBuilder.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+    /*
+     * The set of the MIME types that will be checked for extension mismatches
+     * when checkType is ONLY_MEDIA.
+     * ".jpg", ".jpeg", ".png", ".psd", ".nef", ".tiff", ".bmp", ".tec"
+     * ".aaf", ".3gp", ".asf", ".avi", ".m1v", ".m2v", //NON-NLS
+     * ".m4v", ".mp4", ".mov", ".mpeg", ".mpg", ".mpe", ".mp4", ".rm", ".wmv", ".mpv", ".flv", ".swf"
+     */
+    private static final Set<String> MEDIA_PICS_VIDEO_MIME_TYPES = Stream.of(
+            "image/bmp",
```
[the remark this answers is not in the document]
Comment (briangsweeney) [reply Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesMetaDataBuilder.java]: what does NLS stand for?

**Comment by briangsweeney** [comment]
@APriestman Re NLS stuff - if I understand correctly, no user-facing verbiage should ever be tagged with NLS and should have been implemented using NetBeans bundles (which is what the GUI editor does) and all other strings get the NLS comment?  The NLS comment doesn't do anything at compile time, right?  Is it just there for devs to understand it was intentionally not added to a bundle?

**Comment by APriestman** [comment]
Please leave this PR open and make commits to this branch, if possible.

For codacy, no you don't have to fix all of them. Certainly you don't have to worry about any of the auto-generated and UI-related ones, but ones like "long methods" and "too many fields" are probably fine. I don't think your abstract class necessarily needs to start with "Abstract", either.

File: Core/src/org/sleuthkit/autopsy/datamodel/Md5Node.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                SleuthkitCase tskDb = currentCase.getSleuthkitCase();
+                AbstractFile abstractFile = tskDb.findAllFilesWhere(String.format("obj_id in (%s)", file.getObjectId())).get(0);
+                
+                return new FileInstanceNode(abstractFile, file.getDataSourceName());
+            } catch (NoCurrentCaseException ex) {
+                LOGGER.log(Level.SEVERE, String.format("Unable to create node for file with obj_id: %s.", new Object[]{file.getObjectId()}), ex);
+            } catch (TskCoreException ex) {
+                LOGGER.log(Level.SEVERE, String.format("Unable to create node for file with obj_id: %s.", new Object[]{file.getObjectId()}), ex);
+            }
+            //TODO smells bad...do something?
```
[the remark this answers is not in the document]
Comment (briangsweeney) [reply Core/src/org/sleuthkit/autopsy/datamodel/Md5Node.java]: Turns out that according to the netbeans docs, returning null is acceptable.

File: Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+                        String dataSourceName = resultSet.getString(2);
+                        dataSouceMap.put(objectId, dataSourceName);
+                    }
+                }
+            }
+
+            private void loadImageSources(SleuthkitCase tskDb, Map<Long, String> dataSouceMap) throws SQLException, TskCoreException {
+                //try block releases resources - exceptions are handled in done()
+                try (CaseDbQuery query = tskDb.executeQuery(SELECT_DATA_SOURCES_IMAGE)) {
+                    ResultSet resultSet = query.getResultSet();
```
Comment (APriestman) [inline Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java]: I think I missed this one - please move the ResultSet into the try with resources

Comment (briangsweeney) [reply Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonFilesPanel.java]: I think you caught one, but I misused this paradigm in the same exact way twice.

**Pull request merged**